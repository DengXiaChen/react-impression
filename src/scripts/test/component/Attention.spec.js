import React from 'react';
import { shallow } from 'enzyme';
import { Attention } from '../../components/impression';

const setup = () => {
    const props = {
        theme: 'danger',
        closeable: true,
    };

    const wrapper = shallow(
        <Attention {...props}>
            <Attention.Link href="#">click here</Attention.Link>
        </Attention>
    );

    return {
        props,
        wrapper,
    };
};

describe('Attention', () => {
    const { props, wrapper } = setup();

    it('render', () => {
        expect(wrapper.find('.attention').exists()).toBe(true);
    });

    it('theme', () => {
        expect(wrapper.find(`.attention-${props.theme}`).exists()).toBe(true);
    });

    it('closeable', () => {
        expect(wrapper.find('button.close').exists()).toBe(true);
    });

    it('link', () => {
        expect(wrapper.render().find('.attention-link').prop('href')).toBe('#');
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
