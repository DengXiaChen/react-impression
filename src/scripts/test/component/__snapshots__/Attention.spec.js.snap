// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Attention ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="attention attention-danger"
>
    <AttentionLink
        href="#"
    >
        click here
    </AttentionLink>
    <button
        className="close"
        onClick={[Function]}
        type="button"
    >
        ×
    </button>
</div>,
  "nodes": Array [
    <div
      className="attention attention-danger"
>
      <AttentionLink
            href="#"
      >
            click here
      </AttentionLink>
      <button
            className="close"
            onClick={[Function]}
            type="button"
      >
            ×
      </button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Attention
        closeable={true}
        theme="danger"
>
        <AttentionLink
                href="#"
        >
                click here
        </AttentionLink>
</Attention>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Attention {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "hideHandle": [Function],
        "props": Object {
          "children": <AttentionLink
            href="#"
>
            click here
</AttentionLink>,
          "closeable": true,
          "theme": "danger",
        },
        "refs": Object {},
        "state": Object {
          "show": true,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="attention attention-danger"
>
          <AttentionLink
                    href="#"
          >
                    click here
          </AttentionLink>
          <button
                    className="close"
                    onClick={[Function]}
                    type="button"
          >
                    ×
          </button>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="attention attention-danger"
>
          <AttentionLink
                    href="#"
          >
                    click here
          </AttentionLink>
          <button
                    className="close"
                    onClick={[Function]}
                    type="button"
          >
                    ×
          </button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Attention
    closeable={true}
    theme="danger"
>
    <AttentionLink
        href="#"
    >
        click here
    </AttentionLink>
</Attention>,
}
`;
